### PR TITLE
Fix compile warnings (for 0.99.6-RC6/PR #176)

### PR DESCRIPTION
**Pull request by ldilley**
* Fixed initialization order
* Marked unused variables
* Fixed loop iterator data type
* Removed useless `const` qualifiers
* Removed extraneous whitespace
* Used additional parentheses in comparisons to make intent clearer

### CODE REVIEW

File: source/ai.cpp [hunk 1]
```diff
@@ -68,7 +68,7 @@ bool IsValidAttackTarget( CChar& mChar, CChar *cTarget )
 				// Young players are not valid targets for NPCs outside of dungeons
 				if( cwmWorldState->ServerData()->YoungPlayerSystem() && mChar.IsNpc() && !cTarget->IsNpc() && IsOnline(( *cTarget )) && cTarget->GetAccount().wFlags.test( AB_FLAGS_YOUNG ) && !cTarget->GetRegion()->IsDungeon() )
 				{
-					if( cTarget->GetSocket() && cTarget->GetTimer( tCHAR_YOUNGMESSAGE ) <= cwmWorldState->GetUICurrentTime() || cwmWorldState->GetOverflow() )
+					if( (cTarget->GetSocket() && cTarget->GetTimer( tCHAR_YOUNGMESSAGE ) <= cwmWorldState->GetUICurrentTime()) || cwmWorldState->GetOverflow() )
```
Comment (Xoduz) [inline source/ai.cpp:71]: This changes the flow of the logic 🤔 The check should be:
- "if socket is valid, AND if EITHER tCHAR_YOUNGMESSAGE is lower than GetUICurrentTime OR GetOverflow returns true"

but with this change it becomes:
- "if socket is valid AND tCHAR_YOUNGMESSAGE is lower than GetUICurrentTime, OR if GetOverflow returns true"

File: source/cItem.cpp [hunk 2]
```diff
@@ -2683,8 +2683,8 @@ void CItem::SendToSocket( CSocket *mSock, [[maybe_unused]] bool drawGamePlayer )
 			auto iVisible = GetVisible();
 			auto ownerObj = GetOwnerObj();
 			if(( iVisible != VT_VISIBLE && iVisible != VT_TEMPHIDDEN )
-				|| iVisible == VT_TEMPHIDDEN &&
-				( !ValidateObject( ownerObj ) || ( ValidateObject( ownerObj ) && ownerObj->GetSerial() != mChar->GetSerial() )))
+				|| ((iVisible == VT_TEMPHIDDEN &&
```
Comment (Xoduz) [inline source/cItem.cpp:2686]: This, as well, changes the intended logic. It's supposed to be:
- if visible state is VT_TEMPHIDDEN AND either ownerObj exists OR ownerObj exists and owner serial differs from mChar's serial

File: source/ai.cpp (hunk 1 above)
Comment (ldilley) [reply source/ai.cpp]: Does commit 9d985de now meet the proper criteria?

File: source/cItem.cpp (hunk 2 above)
Comment (ldilley) [reply source/cItem.cpp]: Does commit cf28051 now contain the correct logic?

Comment (Xoduz) [reply source/cItem.cpp]: Not quite 🤔The idea is that the checks on ownerObj & the serial comparison only runs if iVisible equals VT_TEMPHIDDEN. So, maybe... this?
```
if(( iVisible != VT_VISIBLE && iVisible != VT_TEMPHIDDEN )
	|| ( iVisible == VT_TEMPHIDDEN &&
	( !ValidateObject( ownerObj ) || ( ValidateObject( ownerObj ) && ownerObj->GetSerial() != mChar->GetSerial() ))))```

File: source/ai.cpp (hunk 1 above)
Comment (Xoduz) [reply source/ai.cpp]: It does! Thank you :)

File: source/cItem.cpp (hunk 2 above)
Comment (ldilley) [reply source/cItem.cpp]: Fixed in f4e7e35.

**Review by Xoduz — APPROVED** [review]

**Pull request merged**